src/core/dynamic-user.c

```diff
@@ -411,9 +409,9 @@ static int dynamic_user_realize(
                 /* Let's see if a proper, static user or group by this name exists. Try to take the lock on
                  * /etc/passwd, if that fails with EROFS then /etc is read-only. In that case it's fine if we don't
                  * take the lock, given that users can't be added there anyway in this case. */
-                etc_passwd_lock_fd = take_etc_passwd_lock(NULL);
-                if (etc_passwd_lock_fd < 0 && etc_passwd_lock_fd != -EROFS)
-                        return etc_passwd_lock_fd;
+                r = etc_passwd_lock_fd = take_etc_passwd_lock(NULL);
```

Comment:
Hmm?

Reply:
Do you see something wrong?

Reply:
Ah, this is still assigned to etc_passwd_lock. Previously I overlooked that.

We don't usually use r in such cases, but it's OK.